### PR TITLE
Track 2 SDK changes for DNS resolver public preview

### PR DESCRIPTION
**Pull request by arpit-gagneja**
Track 2 SDK changes for DNS resolver public preview

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/dnsresolver/Azure.ResourceManager.DnsResolver/src/Azure.ResourceManager.DnsResolver.csproj [hunk 1]
```diff
@@ -0,0 +1,8 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <Version>1.0.0-beta.1</Version>
+    <PackageId>Azure.ResourceManager.DnsResolver</PackageId>
+    <Description>Azure Resource Manager client SDK for Azure resource provider Microsoft.dnsresolver</Description>
```
Comment (muwaqar) [inline sdk/dnsresolver/Azure.ResourceManager.DnsResolver/src/Azure.ResourceManager.DnsResolver.csproj:5]: fix this

**Review by muwaqar — DISMISSED** [review]

File: sdk/dnsresolver/Azure.ResourceManager.DnsResolver/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Release History
+
+## 1.0.0-preview.1 (Unreleased)
```
Comment (ArthurMa1978) [inline sdk/dnsresolver/Azure.ResourceManager.DnsResolver/CHANGELOG.md:3]: change this to 1.0.0-beta.1

File: sdk/dnsresolver/Azure.ResourceManager.DnsResolver/README.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+This package follows the [new Azure SDK guidelines](https://azure.github.io/azure-sdk/general_introduction.html) which provide a number of core capabilities that are shared amongst all Azure SDKs, including the intuitive Azure Identity library, an HTTP Pipeline with custom policies, error-handling, distributed tracing, and much more.
+
+## Getting started 
+
+### Install the package
+
+Install the Azure dnsresolver management library for .NET with [NuGet](https://www.nuget.org/):
+
+```PowerShell
+Install-Package Azure.ResourceManager.DnsResolver -Version 1.0.0-preview.1
```
Comment (ArthurMa1978) [inline sdk/dnsresolver/Azure.ResourceManager.DnsResolver/README.md:12]: change this to 1.0.0-beta.1

File: sdk/dnsresolver/Azure.ResourceManager.DnsResolver/Directory.Build.props [hunk 4]
```diff
@@ -0,0 +1,6 @@
+﻿<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
Comment (Yao725) [inline sdk/dnsresolver/Azure.ResourceManager.DnsResolver/Directory.Build.props:1]: Can we remove this file?

File: sdk/dnsresolver/Azure.ResourceManager.DnsResolver/README.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
+# Azure dnsresolver Management client library for .NET
```
Comment (Yao725) [inline sdk/dnsresolver/Azure.ResourceManager.DnsResolver/README.md:1]: should be `DnsResolver`.

File: sdk/dnsresolver/Azure.ResourceManager.DnsResolver/src/Azure.ResourceManager.DnsResolver.csproj (hunk 1 above)
Comment (arpit-gagneja) [reply sdk/dnsresolver/Azure.ResourceManager.DnsResolver/src/Azure.ResourceManager.DnsResolver.csproj]: Corrected the description.

File: sdk/dnsresolver/Azure.ResourceManager.DnsResolver/Directory.Build.props (hunk 4 above)
Comment (arpit-gagneja) [reply sdk/dnsresolver/Azure.ResourceManager.DnsResolver/Directory.Build.props]: Removed.

**Comment by arpit-gagneja** [comment]
@ArthurMa1978 @Yao725: I have addressed the comments. Also, I tried to run the pipelines but got an automated reply above saying that I don't have the privilege. Can you please run for me?

File: sdk/dnsresolver/Azure.ResourceManager.DnsResolver/tests/Tests/DnsForwardingRulesetTests.cs [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+            };
+
+            var outboundEndpointName = Recording.GenerateAssetName("outboundEndpoint-");
+            var outboundEndpoint = await dnsResolver.GetOutboundEndpoints().CreateOrUpdateAsync(true, outboundEndpointName, outboundEndpointData);
+
+            outboundEndpointId = outboundEndpoint.Value.Id;
+        }
+
+        [Test]
+        public async Task CreateDnsForwardingRuleset()
```
Comment (Yao725) [inline sdk/dnsresolver/Azure.ResourceManager.DnsResolver/tests/Tests/DnsForwardingRulesetTests.cs]: Can we add the attribute `RecordedTest` to all the test cases?

File: sdk/dnsresolver/Azure.ResourceManager.DnsResolver/tests/DnsResolverTestBase.cs [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+                {
+                    Tags =
+                    {
+                        { "test", "env" }
+                    }
+                });
+            return rgOp.Value;
+        }
+
+        protected async Task CreateVirtualNetworkAsync(string virtualNetworkName)
```
Comment (Yao725) [inline sdk/dnsresolver/Azure.ResourceManager.DnsResolver/tests/DnsResolverTestBase.cs:57]: Can we add a project reference to `Azure.ResourceManager.Network` and use it to create the vnet?

File: sdk/dnsresolver/Azure.ResourceManager.DnsResolver/tests/DnsResolverTestBase.cs [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+                        addressPrefixes = new[] { "10.0.0.0/8" },
+                    }
+                },
+            };
+
+            var httpContent = new StringContent(JsonConvert.SerializeObject(virtualNetwork), Encoding.UTF8, "application/json");
+            await client.PutAsync(baseUri + relativeUri, httpContent);
+        }
+
+        protected async Task CreateSubnetAsync(string virtualNetworkName)
```
Comment (Yao725) [inline sdk/dnsresolver/Azure.ResourceManager.DnsResolver/tests/DnsResolverTestBase.cs:79]: Same as above

File: sdk/dnsresolver/Azure.ResourceManager.DnsResolver/tests/DnsResolverTestBase.cs (hunk 7 above)
Comment (arpit-gagneja) [reply sdk/dnsresolver/Azure.ResourceManager.DnsResolver/tests/DnsResolverTestBase.cs:57]: This is intentional. For recording, we have selected the ARM dogfood environment as we don't want to create so many resources in the prod environment. As of today, there is a limitation that virtual networks and subnets can't be created in the ARM dogfood. As a workaround, my team has written the simulator service to create VNETs and subnets. This function is calling that simulator service.

File: sdk/dnsresolver/Azure.ResourceManager.DnsResolver/tests/DnsResolverTestBase.cs (hunk 8 above)
Comment (arpit-gagneja) [reply sdk/dnsresolver/Azure.ResourceManager.DnsResolver/tests/DnsResolverTestBase.cs:79]: Replied to the above comment.

File: sdk/dnsresolver/Azure.ResourceManager.DnsResolver/tests/Tests/DnsForwardingRulesetTests.cs (hunk 6 above)
Comment (arpit-gagneja) [reply sdk/dnsresolver/Azure.ResourceManager.DnsResolver/tests/Tests/DnsForwardingRulesetTests.cs:67]: Update the attribute.

**Review by Yao725 — APPROVED** [review]

**Pull request merged**